MSAL/test/unit/MSALB2CPolicyTests.m

```diff
@@ -0,0 +1,148 @@
+                                            query:@"p=b2c_1_policy"];
+    
+    // User identifier should be uid-policy
+    MSALTestURLResponse *tokenResponse =
+    [MSALTestURLResponse authCodeResponse:@"i am an auth code"
+                                authority:@"https://login.microsoftonline.com/contosob2c"
+                                    query:@"p=b2c_1_policy"
+                                   scopes:[NSOrderedSet orderedSetWithArray:@[@"fakeb2cscopes", @"openid", @"profile", @"offline_access"]]
+                               clientInfo:@{ @"uid" : @"1-b2c_1_policy", @"utid" : [MSALTestIdTokenUtil defaultTenantId]}];
+    
+    [MSALTestURLSession addResponses:@[oidcResponse, tokenResponse]];
+    
+    [MSALTestSwizzle classMethod:@selector(startWebUIWithURL:context:completionBlock:)
+                           class:[MSALWebUI class]
+                           block:(id)^(id obj, NSURL *url, id<MSALRequestContext>context, MSALWebUICompletionBlock completionBlock)
+     {
+         (void)obj;
+         (void)context;
+         
+         XCTAssertNotNil(url);
+         
+         // State preserving and url are tested separately
+         NSDictionary *QPs = [NSDictionary msalURLFormDecode:url.query];
+         NSString *state = QPs[@"state"];
+         
+         NSString *responseString = [NSString stringWithFormat:UNIT_TEST_DEFAULT_REDIRECT_URI"?code=%@&state=%@", @"i+am+an+auth+code", state];
+         completionBlock([NSURL URLWithString:responseString], nil);
+     }];
+    
+    NSError *error = nil;
+    
+    // Create application object with the first policy as authority
+    MSALPublicClientApplication *application =
+    [[MSALPublicClientApplication alloc] initWithClientId:UNIT_TEST_CLIENT_ID
+                                                authority:firstAuthority
+                                                    error:&error];
+    
+    application.tokenCache = cacheUtil.cache;
+    
+    XCTAssertNotNil(application);
+    XCTAssertNil(error);
+    
+    __block dispatch_semaphore_t dsem = dispatch_semaphore_create(0);
+    
+    [application acquireTokenForScopes:@[@"fakeb2cscopes"]
+                       completionBlock:^(MSALResult *result, NSError *error)
+     {
+         XCTAssertNil(error);
+         XCTAssertNotNil(result);
+         
+         NSString *userIdentifier = [NSString stringWithFormat:@"1-b2c_1_policy.%@", [MSALTestIdTokenUtil defaultTenantId]];
+         XCTAssertEqualObjects(result.user.userIdentifier, userIdentifier);
+         dispatch_semaphore_signal(dsem);
+     }];
+    
+    while (dispatch_semaphore_wait(dsem, DISPATCH_TIME_NOW))
+    {
+        [[NSRunLoop mainRunLoop] runMode:NSDefaultRunLoopMode beforeDate: [NSDate distantFuture]];
+    }
+    
+    // Now acquiretoken call with second policy (b2c_2_policy)
+    NSString *secondAuthority = @"https://login.microsoftonline.com/tfp/contosob2c/b2c_2_policy";
```

Comment:
so this pattern really calls for a helper method. we're basically repeating the same block twice with slightly different parameters, right?